<center>
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/SN_web_lightmode.png" width="300" alt="cognitiveclass.ai logo">
</center>

<h1 align=center><font size = 5>Assignment: Notebook for Graded Assessment</font></h1>


# Introduction

Using this Python notebook you will:

1.  Understand three Chicago datasets
2.  Load the three datasets into three tables in a SQLIte database
3.  Execute SQL queries to answer assignment questions


## Understand the datasets

To complete the assignment problems in this notebook you will be using three datasets that are available on the city of Chicago's Data Portal:

1.  <a href="https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Socioeconomic Indicators in Chicago</a>
2.  <a href="https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Public Schools</a>
3.  <a href="https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Crime Data</a>

### 1. Socioeconomic Indicators in Chicago

This dataset contains a selection of six socioeconomic indicators of public health significance and a “hardship index,” for each Chicago community area, for the years 2008 – 2012.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2](https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 2. Chicago Public Schools

This dataset shows all school level performance data used to create CPS School Report Cards for the 2011-2012 school year. This dataset is provided by the city of Chicago's Data Portal.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t](https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 3. Chicago Crime Data

This dataset reflects reported incidents of crime (with the exception of murders where data exists for each victim) that occurred in the City of Chicago from 2001 to present, minus the most recent seven days.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2](https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)


Execute the below code cell to avoid prettytable default error.


In [1]:
#!pip install ipython-sql prettytable

import prettytable

prettytable.DEFAULT = 'DEFAULT'

### Store the datasets in database tables

To analyze the data using SQL, it first needs to be loaded into SQLite DB.
We will create three tables in as under:

1.  **CENSUS_DATA**
2.  **CHICAGO_PUBLIC_SCHOOLS**
3.  **CHICAGO_CRIME_DATA**


Load the `pandas` and `sqlite3` libraries and establish a connection to `FinalDB.db`


### Download the datasets

This assignment requires you to have these three tables populated with a subset of the whole datasets.

In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet. 

Use the links below to read the data files using the Pandas library. 

* Chicago Census Data

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

* Chicago Public Schools

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

* Chicago Crime Data

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

**NOTE:** Ensure you use the datasets available on the links above instead of directly from the Chicago Data Portal. The versions linked here are subsets of the original datasets and have some of the column names modified to be more database friendly which will make it easier to complete this assignment.


In [2]:
import pandas as pd
import sqlite3

conn = sqlite3.connect('FinalDB.db')

Load the SQL magic module


In [3]:
%load_ext sql

Use `Pandas` to load the data available in the links above to dataframes. Use these dataframes to load data on to the database `FinalDB.db` as required tables.


In [4]:
df = pd.read_csv('https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01')
df.to_sql('Chicago_Census_Data', conn, if_exists='replace', index = False)

78

In [5]:
df = pd.read_csv('https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01')
df.to_sql('Chicago_Public_Schools', conn, if_exists='replace', index = False)

566

In [6]:
df = pd.read_csv('https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01')
df.to_sql('Chicago_Crime_Data', conn, if_exists='replace', index = False)

533

Establish a connection between SQL magic module and the database `FinalDB.db`


In [7]:
%sql sqlite:///FinalDB.db

In [8]:
%sql SELECT name FROM sqlite_master WHERE type='table';

 * sqlite:///FinalDB.db
Done.


[('Chicago_Census_Data',),
 ('Chicago_Public_Schools',),
 ('Chicago_Crime_Data',)]

In [18]:
%%sql
SELECT * FROM Chicago_Census_Data LIMIT 3;

 * sqlite:///FinalDB.db
Done.


[(1.0, 'Rogers Park', 7.7, 23.6, 8.7, 18.2, 27.5, 23939, 39.0),
 (2.0, 'West Ridge', 7.8, 17.2, 8.8, 20.8, 38.5, 23040, 46.0),
 (3.0, 'Uptown', 3.8, 24.0, 8.9, 11.8, 22.2, 35787, 20.0)]

In [19]:
%%sql
SELECT * FROM Chicago_Public_Schools LIMIT 3;

 * sqlite:///FinalDB.db
Done.


[(610038, 'Abraham Lincoln Elementary School', 'ES', '615 W Kemper Pl', 'Chicago', 'IL', 60614, '(773) 534-5720', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610038.pdf', 'Fullerton Elementary Network', 'NORTH-NORTHWEST SIDE COLLABORATIVE', 'No', 'Standard', 'Not on Probation', 'Level 1', 'Yes', 'Very Strong', 99.0, 'Very Strong', '99', 'Strong', 74.0, 'Strong', 66.0, 'Weak', '65', 'Strong', '70', 'Strong', '56', 'Average', '47', '96.00%', 2.0, '96.40%', '95.80%', '80.1', '43.3', '89.6', '84.9', '60.7', '62.6', '81.9', '85.2', '52', '62.4', '66.3', '77.9', 69.7, 64.4, 0.2, 0.9, 'Yellow', 'Green', '67.1', '54.5', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 813, 33, 'NDA', 1171699.458, 1915829.428, 41.92449696, -87.64452163, 7, 'LINCOLN PARK', 43, 18, '(41.92449696, -87.64452163)'),
 (610281, 'Adam Clayton Powell Paideia Community Academy Elementary School', 'ES', '7511 S South Shore Dr', 'Chicago', 'IL', 60649, '(773) 535-6650', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610281.pdf', 'Skyway Elementary Network', 'SOUTH SIDE COLLABORATIVE', 'No', 'Track_E', 'Not on Probation', 'Level 1', 'No', 'Average', 54.0, 'Strong', '66', 'Strong', 74.0, 'Very Strong', 84.0, 'Weak', '63', 'Strong', '76', 'Weak', '46', 'Average', '50', '95.60%', 15.7, '95.30%', '100.00%', '62.4', '51.7', '21.9', '15.1', '29', '42.8', '38.5', '27.4', '44.8', '42.7', '14.1', '34.4', 16.8, 16.5, 0.7, 1.4, 'Green', 'Green', '17.2', '27.3', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 521, 46, 'NDA', 1196129.985, 1856209.466, 41.76032435, -87.55673627, 43, 'SOUTH SHORE', 7, 4, '(41.76032435, -87.55673627)'),
 (610185, 'Adlai E Stevenson Elementary School', 'ES', '8010 S Kostner Ave', 'Chicago', 'IL', 60652, '(773) 535-2280', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610185.pdf', 'Midway Elementary Network', 'SOUTHWEST SIDE COLLABORATIVE', 'No', 'Standard', 'Not on Probation', 'Level 2', 'No', 'Strong', 61.0, 'NDA', 'NDA', 'Average', 50.0, 'Weak', 36.0, 'Weak', 'NDA', 'NDA', 'NDA', 'Average', '47', 'Weak', '41', '95.70%', 2.3, '94.70%', '98.30%', '53.7', '26.6', '38.3', '34.7', '43.7', '57.3', '48.8', '39.2', '46.8', '44', '7.5', '21.9', 18.3, 15.5, -0.9, -1.0, 'Red', 'Red', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 1324, 44, 'NDA', 1148427.165, 1851012.215, 41.74711093, -87.73170248, 70, 'ASHBURN', 13, 8, '(41.74711093, -87.73170248)')]

In [20]:
%%sql
SELECT * FROM Chicago_Crime_Data LIMIT 3;

 * sqlite:///FinalDB.db
Done.


[(3512276, 'HK587712', '2004-08-28', '047XX S KEDZIE AVE', '890', 'THEFT', 'FROM BUILDING', 'SMALL RETAIL STORE', 0, 0, 911, 9, 14.0, 58.0, '6', 1155838.0, 1873050.0, 2004, 41.8074405, -87.70395585, '(41.8074405, -87.703955849)'),
 (3406613, 'HK456306', '2004-06-26', '009XX N CENTRAL PARK AVE', '820', 'THEFT', '$500 AND UNDER', 'OTHER', 0, 0, 1112, 11, 27.0, 23.0, '6', 1152206.0, 1906127.0, 2004, 41.89827996, -87.71640551, '(41.898279962, -87.716405505)'),
 (8002131, 'HT233595', '2011-04-04', '043XX S WABASH AVE', '820', 'THEFT', '$500 AND UNDER', 'NURSING HOME/RETIREMENT HOME', 0, 0, 221, 2, 3.0, 38.0, '6', 1177436.0, 1876313.0, 2011, 41.81593313, -87.62464213, '(41.815933131, -87.624642127)')]

You can now proceed to the the following questions. Please note that a graded assignment will follow this lab and there will be a question on each of the problems stated below. It can be from the answer you received or the code you write for this problem. Therefore, please keep a note of both your codes as well as the response you generate.


## Problems

Now write and execute SQL queries to solve assignment problems

### Problem 1

##### Find the total number of crimes recorded in the CRIME table.


In [17]:
%sql SELECT count(*) \
    FROM Chicago_Crime_Data;

 * sqlite:///FinalDB.db
Done.


[(533,)]

### Problem 2

##### List community area names and numbers with per capita income less than 11000.


In [21]:
%sql SELECT Community_Area_Number, Community_Area_Name \
    FROM Chicago_Census_Data \
    WHERE Per_Capita_Income < 11000;

 * sqlite:///FinalDB.db
Done.


[(26.0, 'West Garfield Park'),
 (30.0, 'South Lawndale'),
 (37.0, 'Fuller Park'),
 (54.0, 'Riverdale')]

### Problem 3

##### List all case numbers for crimes involving minors?(children are not considered minors for the purposes of crime analysis) 


In [24]:
%sql SELECT Description \
    FROM Chicago_Crime_Data;

 * sqlite:///FinalDB.db
Done.


[('FROM BUILDING',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('FINANCIAL ID THEFT: OVER $300',),
 ('$500 AND UNDER',),
 ('OVER $500',),
 ('OVER $500',),
 ('RETAIL THEFT',),
 ('OVER $500',),
 ('$500 AND UNDER',),
 ('RETAIL THEFT',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('AGG: FINANCIAL ID THEFT',),
 ('FROM BUILDING',),
 ('$500 AND UNDER',),
 ('RETAIL THEFT',),
 ('$500 AND UNDER',),
 ('OVER $500',),
 ('PURSE-SNATCHING',),
 ('RETAIL THEFT',),
 ('RETAIL THEFT',),
 ('$500 AND UNDER',),
 ('RETAIL THEFT',),
 ('OVER $500',),
 ('$500 AND UNDER',),
 ('POCKET-PICKING',),
 ('$500 AND UNDER',),
 ('OVER $500',),
 ('OVER $500',),
 ('FROM BUILDING',),
 ('OVER $500',),
 ('RETAIL THEFT',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('OVER $500',),
 ('ATTEMPT THEFT',),
 ('OVER $500',),
 ('$500 AND UNDER',),
 ('RETAIL THEFT',),
 ('$500 AND UNDER',),
 ('FROM BUILDING',),
 ('RETAIL THEFT',),
 ('$500 AND UNDER',),
 ('OVER $500',),
 ('$500 AND UNDER',),
 ('OVER $500',),
 ('FROM BUILDING',),
 ('OVER $500',),
 ('FROM BUILDING',),
 ('OVER $500',),
 ('FROM BUILDING',),
 ('$500 AND UNDER',),
 ('OVER $500',),
 ('OVER $500',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('OVER $500',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('RETAIL THEFT',),
 ('FROM BUILDING',),
 ('$500 AND UNDER',),
 ('OVER $500',),
 ('FROM BUILDING',),
 ('$500 AND UNDER',),
 ('FROM BUILDING',),
 ('FROM BUILDING',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('RETAIL THEFT',),
 ('OVER $500',),
 ('$500 AND UNDER',),
 ('OVER $500',),
 ('OVER $500',),
 ('$500 AND UNDER',),
 ('OVER $500',),
 ('$500 AND UNDER',),
 ('RETAIL THEFT',),
 ('FROM BUILDING',),
 ('RETAIL THEFT',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('FROM BUILDING',),
 ('OVER $500',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('OVER $500',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('OVER $500',),
 ('OVER $500',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('FROM BUILDING',),
 ('$500 AND UNDER',),
 ('$500 AND UNDER',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('AGG PO HANDS NO/MIN INJURY',),
 ('PRO EMP HANDS NO/MIN INJURY',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('AGGRAVATED:KNIFE/CUTTING INSTR',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('SIMPLE',),
 ('AGGRAVATED: OTHER DANG WEAPON',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('AGGRAVATED:KNIFE/CUTTING INSTR',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('AGGRAVATED: OTHER DANG WEAPON',),
 ('SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('AGGRAVATED: OTHER DANG WEAPON',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('SIMPLE',),
 ('AGGRAVATED: OTHER DANG WEAPON',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('AGGRAVATED: HANDGUN',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('AGGRAVATED: HANDGUN',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('AGGRAVATED DOMESTIC BATTERY',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('SIMPLE',),
 ('AGGRAVATED: OTHER DANG WEAPON',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('DOM

In [49]:
%sql SELECT Case_Number \
    FROM Chicago_Crime_Data \
    WHERE description LIKE '%MINOR%';

 * sqlite:///FinalDB.db
Done.


[('HL266884',), ('HK238408',)]

### Problem 4

##### List all kidnapping crimes involving a child?


In [54]:
%sql SELECT Case_Number, Description\
    FROM Chicago_Crime_Data \
    WHERE PRIMARY_TYPE LIKE "%kidnapping%" AND description LIKE "%child%";

 * sqlite:///FinalDB.db
Done.


[('HN144152', 'CHILD ABDUCTION/STRANGER')]

### Problem 5

##### List the kind of crimes that were recorded at schools. (No repetitions)


In [35]:
%sql SELECT LOCATION_DESCRIPTION, PRIMARY_TYPE \
    FROM Chicago_Crime_Data \
    WHERE LOCATION_DESCRIPTION LIKE "%SCHOOL%" \
        GROUP BY PRIMARY_TYPE;

 * sqlite:///FinalDB.db
Done.


[('SCHOOL, PUBLIC, GROUNDS', 'ASSAULT'),
 ('SCHOOL, PUBLIC, GROUNDS', 'BATTERY'),
 ('SCHOOL, PUBLIC, GROUNDS', 'CRIMINAL DAMAGE'),
 ('SCHOOL, PUBLIC, GROUNDS', 'CRIMINAL TRESPASS'),
 ('SCHOOL, PUBLIC, GROUNDS', 'NARCOTICS'),
 ('SCHOOL, PRIVATE, BUILDING', 'PUBLIC PEACE VIOLATION')]

### Problem 6

##### List the type of schools along with the average safety score for each type.


In [40]:
%sql SELECT "Elementary, Middle, or High School" AS TYPE_OF_SCHOOL, round(AVG(SAFETY_SCORE), 2) AS "Avg Safety Score"\
    FROM Chicago_Public_Schools \
    GROUP BY "Elementary, Middle, or High School";

 * sqlite:///FinalDB.db
Done.


[('ES', 49.52), ('HS', 49.62), ('MS', 48.0)]

### Problem 7

##### List 5 community areas with highest % of households below poverty line


In [ ]:
%sql SELECT Community_Area_Number, Community_Area_Name, PERCENT_HOUSEHOLDS_BELOW_POVERTY AS "Households Below Poverty Line" \
    FROM Chicago_Census_Data \
    ORDER BY PERCENT_HOUSEHOLDS_BELOW_POVERTY DESC \
    LIMIT 5;

 * sqlite:///FinalDB.db
Done.


[(54.0, 'Riverdale', 56.5),
 (37.0, 'Fuller Park', 51.2),
 (68.0, 'Englewood', 46.6),
 (29.0, 'North Lawndale', 43.1),
 (27.0, 'East Garfield Park', 42.4)]

### Problem 8

##### Which community area is most crime prone? Display the coumminty area number only.


In [45]:
%sql SELECT Community_Area_Number \
    FROM Chicago_Crime_Data \
    GROUP BY Community_Area_Number \
    ORDER BY count(*) DESC \
    LIMIT 1;

 * sqlite:///FinalDB.db
Done.


[(25.0,)]


Double-click **here** for a hint

<!--
Query for the 'community area number' that has most number of incidents
-->


### Problem 9

##### Use a sub-query to find the name of the community area with highest hardship index


In [46]:
%sql SELECT Community_Area_Name \
    FROM Chicago_Census_Data \
        WHERE Hardship_Index = (SELECT MAX(Hardship_Index) FROM Chicago_Census_Data);

 * sqlite:///FinalDB.db
Done.


[('Riverdale',)]

### Problem 10

##### Use a sub-query to determine the Community Area Name with most number of crimes?


In [47]:
%sql SELECT Community_Area_Name \
    FROM Chicago_Census_Data \
        WHERE Community_Area_Number = (SELECT Community_Area_Number \
        FROM Chicago_Crime_Data \
        GROUP BY Community_Area_Number \
        ORDER BY count(*) DESC \
        LIMIT 1);

 * sqlite:///FinalDB.db
Done.


[('Austin',)]

## Author(s)

<h4> Hima Vasudevan </h4>
<h4> Rav Ahuja </h4>
<h4> Ramesh Sannreddy </h4>

## Contribtuor(s)

<h4> Malika Singla </h4>
<h4>Abhishek Gagneja</h4>
<!--
## Change log

| Date       | Version | Changed by        | Change Description                             |
| ---------- | ------- | ----------------- | ---------------------------------------------- |
|2023-10-18  | 2.6     | Abhishek Gagneja  | Modified instruction set |
| 2022-03-04 | 2.5     | Lakshmi Holla     | Changed markdown.                   |
| 2021-05-19 | 2.4     | Lakshmi Holla     | Updated the question                           |
| 2021-04-30 | 2.3     | Malika Singla     | Updated the libraries                          |
| 2021-01-15 | 2.2     | Rav Ahuja         | Removed problem 11 and fixed changelog         |
| 2020-11-25 | 2.1     | Ramesh Sannareddy | Updated the problem statements, and datasets   |
| 2020-09-05 | 2.0     | Malika Singla     | Moved lab to course repo in GitLab             |
| 2018-07-18 | 1.0     | Rav Ahuja         | Several updates including loading instructions |
| 2018-05-04 | 0.1     | Hima Vasudevan    | Created initial version                        |
-->
## <h3 align="center"> © IBM Corporation 2023. All rights reserved. <h3/>
